velox/expression/CastExpr-inl.h

```diff
@@ -268,7 +268,12 @@ void CastExpr::applyToSelectedNoThrowLocal(
     rows.template applyToSelected([&](auto row) INLINE_LAMBDA {
       try {
         func(row);
-      } catch (...) {
+      } catch (const VeloxException& e) {
+        if (!e.isUserError()) {
+          throw;
```

Comment:
So if its a LoggedException OR VeloxRuntimeError we will throw , everything else is suppressed ? (based on catch below).

Reply:
Yes, VeloxUserError and everything that's not VeloxException are suppressed. 

Presto suppress all kinds of errors: https://github.com/prestodb/presto/blob/d3393a2bb0dcc562fc873d332c5142dde9ade066/presto-main/src/main/java/com/facebook/presto/operator/scalar/TryCastFunction.java#L96-L97. But I don't think cast would throw runtime error in Presto. We, however, need cast to throw runtime error to trigger the retry.

LoggedException in Velox is only used in dwio code, so it's irrelevant.